src/__test__/api.test.ts

```diff
@@ -0,0 +1,130 @@
+import { API } from '../api'
+
+describe('API', () => {
+  const apiKey = 'YOUR_API_KEY'
+  const endpoint = 'https://example.com'
+  const path = '/api'
+  const headers = { 'X-Header': 'value' }
+
+  let api: API
+
+  beforeEach(() => {
+    api = new API(apiKey, endpoint, path, headers, true)
+  })
+
+  afterEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  describe('request', () => {
+    it('should make a POST request and return response data', async () => {
+      const mockResponse = { data: 'Response data' }
+      global.fetch = jest.fn().mockResolvedValueOnce({
+        ok: true,
+        json: jest.fn().mockResolvedValueOnce(mockResponse),
+      })
+
+      const method = 'POST'
+      const url = `${endpoint}${path}`
+      const body = { param: 'value' }
+
+      const result = await api['request'](method, url, body)
+
+      expect(global.fetch).toHaveBeenCalledWith(
```

Comment:
entirely up to you but in the integration tests I use nock (https://github.com/nock/nock) which gives alot of control rather than mocking fetch. You can simulate network errors and delays well.